src/main/java/com/trillia/entities/Contact.java

```diff
@@ -0,0 +1,107 @@
+package com.trillia.entities;
+
+import org.hibernate.search.annotations.*;
+import org.hibernate.search.annotations.Index;
+
+import javax.persistence.*;
+import javax.xml.bind.annotation.XmlElement;
+import java.io.Serializable;
+
+/**
+ * Created by mmathew on 27/05/16.
+ */
+
+
+@Entity
+@Table(name = "trl_contacts", uniqueConstraints = {@UniqueConstraint(columnNames = {"tenantId"}), @UniqueConstraint(columnNames = {"contactAlias"})})
+@Indexed(index = "Contacts")
+
+public class Contact implements Serializable {
+
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(name = "contact_id", nullable = false)
+    @XmlElement(name = "contact-id")
+    private long id;
+
+
+    @XmlElement(name = "tenant-id")
+    @Fields({
+            @Field,
+            @Field(analyze = Analyze.NO, store = Store.YES)
+    })
+
+    @OneToOne(fetch = FetchType.LAZY, mappedBy = "tenant_id", cascade = CascadeType.ALL)
+    private Subscriber tenantId;
+
+
+    @XmlElement(name = "contact-alias")
+    @Column(name = "contact_alias")
+
+    @Fields({
+            @Field,
+            @Field(index = Index.YES, analyze = Analyze.NO, store = Store.YES)
+    })
+    private String contactAlias;
+
+
+    @XmlElement(name = "contact-type")
+    @Column(name = "contact_type")
+
+    @Fields({
+            @Field,
+            @Field(index = Index.YES, analyze = Analyze.NO, store = Store.YES)
+    })
+    private String contactType;
+
+
+    @XmlElement(name = "contact")
+    @Column(name = "contact")
+
+    @Fields({
+            @Field,
+            @Field(index = Index.YES, analyze = Analyze.NO, store = Store.YES)
+    })
+    private String contact;
+
```

Comment:
As you mentioned in the beginning , you may try to keep tenantId and revisioncontrolbean in a base class and extend that class. I faced some issue at some point, which I can't recollect now. Worth giving it a try.